streams/src/main/java/org/apache/kafka/streams/processor/ProcessorRecordContext.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor;
+
+/**
+ * An {@link RecordContext} that also can forward to children of
+ * the current {@link org.apache.kafka.streams.processor.internals.ProcessorNode}
+ */
+public interface ProcessorRecordContext extends RecordContext {
```

Comment:
`ProcessorContextImpl.forward(..)` used to call `task.forward(..)` - this is not valid anymore as when an item is flushed from the cash, the `context` changes and we can't simply call `task.forward`, we need to set the new `ProcessorRecordContext` on the context. The `ProcessorRecordContext` contains the correct `ProcessorNode` to resume processing of the topology. If we don't do this, we end up forwarding to a random node in the topology or no node at all.